core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/BasicLoadBalancingPolicy.java

```diff
@@ -288,14 +292,21 @@ protected Set<Node> getReplicas(@Nullable Request request, @Nullable Session ses
     if (!maybeTokenMap.isPresent()) {
       return Collections.emptySet();
     }
+    TabletMap tabletMap = context.getMetadataManager().getMetadata().getTabletMap();
```

Comment:
When can it happen that `maybeTokenMap` is not present?
If it is happen we return always, even if table is tablet-based (in which case we have no need for `TokenMap`).
In Rust driver we split similar method into 2 branches of a big if - one for tablets and one for vnodes.